### PR TITLE
[dashing] Set allocator before goto fail.

### PR DESCRIPTION
**Pull request by BorjaOuterelo**
Fix https://github.com/ros2/rcl/issues/539

It should be added also in master and Eloquent

### CODE REVIEW

File: rcl/src/rcl/init.c [hunk 1]
```diff
@@ -82,6 +82,9 @@ rcl_init(
   // Zero initialize rmw context first so its validity can by checked in cleanup.
   context->impl->rmw_context = rmw_get_zero_initialized_context();
 
+  // Store the allocator.
+  context->impl->allocator = allocator;
```
Comment (jacobperron) [inline rcl/src/rcl/init.c]: The allocator is also stored below (L154). I think we should remove that redundant logic.

Comment (BorjaOuterelo) [reply rcl/src/rcl/init.c]: Right, I already fixed it.

**Comment by jacobperron** [comment]
@BorjaOuterelo I just noticed that this PR is targeting `dashing`. It would be better to target `master` and then we can backport the changes for Eloquent and Dashing.

**Comment by jacobperron** [comment]
I've ported the change to the `master` branch in https://github.com/ros2/rcl/pull/546

**Comment by BorjaOuterelo** [comment]
Shall I be worried for the CI fail?
http://build.ros2.org/job/Epr__rcl__ubuntu_bionic_amd64/77/ It seems more like an infrastructure issue.

**Comment by jacobperron** [comment]
> Shall I be worried for the CI fail?

No. The failing job is for Eloquent (it was accidentally triggered when I temporary changed the target of this PR).

**Comment by jacobperron** [comment]
I'm waiting for #546 to be merged before this one.

**Pull request merged**